.github/workflows/main.yml

```diff
@@ -56,6 +61,40 @@ jobs:
             type=semver,pattern={{version}}
             type=semver,pattern={{major}}.{{minor}}
 
+      - name: Wait for base image if needed
+        if: github.event_name == 'pull_request'
+        run: |
+          # Wait for base image workflow to complete if base_image directory was modified
+          if git diff --name-only origin/${{ github.base_ref }} | grep -q "^base_image/"; then
+            echo "Base image changes detected, waiting for parallel-base-image workflow to complete..."
+            attempt=1
+            max_attempts=30
+            while [ $attempt -le $max_attempts ]; do
+              status=$(curl -s -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+                "https://api.github.com/repos/${{ github.repository }}/actions/runs?event=pull_request&branch=${{ github.head_ref }}&status=completed" | \
+                jq -r '.workflow_runs[] | select(.name == "Build and Push Base Docker image (parallel)") | .conclusion')
+
+              if [ "$status" = "success" ]; then
+                echo "Base image workflow completed successfully"
+                break
+              elif [ "$status" = "failure" ]; then
+                echo "Base image workflow failed"
+                exit 1
+              fi
+
+              echo "Waiting for base image workflow to complete (attempt $attempt/$max_attempts)..."
+              sleep 20
```

Comment:
20 seconds * 30 max_attempts = 10 minutes but I think it usually takes longer. For example, [today's build](https://github.com/r2dt-bio/R2DT/actions/runs/12446592868/usage) took 32m  24s.